# 18 — De Sitter Cavitation Engine

**No Singularity: the Abrikosov-Vortex Core and De Sitter Cavitation over a Black Hole's Life.**

`ValaQuenta/modules/desitter_cavitation/` · `DeSitterCavitationModule` · v0.100 · confidence floor THEORETICAL.

**Calculation, not simulation** — closed-form scalars, ratios, timescales. Exercises the whole formulary and shows the open items honestly.

Fourth Age paper: `FourthAgePapers/DeSitterCavitation/`.

In [1]:
import sys; import os; sys.path.insert(0, os.path.abspath('../../..'))
from ValaQuenta.modules.desitter_cavitation.tools import DeSitterCavitationModule
from ValaQuenta.modules.desitter_cavitation import maths as M
E = DeSitterCavitationModule()
print(E.summary())

Module: De Sitter Cavitation Engine — No Singularity: the Abrikosov-Vortex Core
Version: 0.100
Description: The black-hole interior is a finite, sub-Planckian de Sitter core — the Abrikosov vortex core made gravitational: the condensate goes to zero (a Riemann zero, winding 1) while density, pressure and curvature stay finite. HOLCUS: the maximum curvature is the de Sitter Kretschmann scalar at L_dS = r_s, K_core(M) = (3/2) c^8 / (G^4 M^4) — mass-dependent, M^-4, sub-Planckian for every M > (3/2)^(1/4) m_Pl, with a ringdown-echo delay ~ r_s/c as its observational shadow. The core releases stiff space (Lambda-signed) and stiff matter (radiative) over the hole's life and unwraps at evaporation — the De Sitter Cavitation. Falsifier: a divergent core curvature, or one pinned to K_Planck independent of M.
Confidence floor: THEORETICAL
Equations: 8
  [◈][✓] full_desitter_cavitation: The whole engine: Holcus + no-singularity check + mass-class table + partition + cosmic budget
  [✓][✓] kretsc

## The Holcus prediction

`K_core(M) = 24 / L_dS^4 = 24 / r_s^4 = (3/2) c^8 / (G^4 M^4)` — the de Sitter Kretschmann scalar at `L_dS = r_s`. Finite for all `M>0`, scales as `M^-4`, sub-Planckian for every `M > (3/2)^(1/4) m_Pl`.

In [2]:
Msun = M.M_SUN
for label, Mkg in [('kugelblitz 1e12 kg', 1e12), ('stellar 10 Msun', 10*Msun),
                   ('IMBH 1e4 Msun', 1e4*Msun), ('SMBH 1e9 Msun', 1e9*Msun)]:
    r = E.run('kretschmann_core', {'M_kg': Mkg})['result']
    print(f"{label:20s}  K_core={r['K_core_m^-4']:.3e} m^-4   "
          f"K/K_Planck={r['K_core_over_K_Planck']:.2e}   "
          f"closed-form matches={r['closed_form_matches']}")
print()
print('Planck-mass crossover:', M.planck_mass_crossover())

kugelblitz 1e12 kg    K_core=4.932e+60 m^-4   K/K_Planck=3.37e-79   closed-form matches=True
stellar 10 Msun       K_core=3.152e-17 m^-4   K/K_Planck=2.15e-156   closed-form matches=True
IMBH 1e4 Msun         K_core=3.152e-29 m^-4   K/K_Planck=2.15e-168   closed-form matches=True
SMBH 1e9 Msun         K_core=3.152e-49 m^-4   K/K_Planck=2.15e-188   closed-form matches=True

Planck-mass crossover: {'M_crossover_kg': 2.408620535762847e-08, 'M_crossover_over_mPl': 1.1066819197003215, 'note': 'K_core = K_Planck only here; M > this ⟹ sub-Planckian core'}


## The no-singularity check

A consistency scorecard — **not** a proof that nature picks this interior. It verifies: (a) `K_core` finite everywhere, (b) `K_core ∝ M^-4`, (c) sub-Planckian above the crossover, and the contrast (d) Schwarzschild `K(r) → ∞` as `r → 0`.

In [3]:
chk = E.run('no_singularity_check', {})['result']
for k in ('finite','scales_as_M_minus_4','sub_planckian_above_crossover',
          'schwarzschild_diverges_at_r0','PASS'):
    print(f'  {k:32s} {chk[k]}')
print()
print('  Schwarzschild K(r) contrast (10 Msun): r [m], K [m^-4]')
for r, K in chk['schwarzschild_contrast']:
    print(f'    r={r:.3e}  K={K:.3e}')

  finite                           True
  scales_as_M_minus_4              True
  sub_planckian_above_crossover    True
  schwarzschild_diverges_at_r0     True
  PASS                             True

  Schwarzschild K(r) contrast (10 Msun): r [m], K [m^-4]
    r=2.954e+04  K=1.576e-17
    r=2.954e+01  K=1.576e+01
    r=2.954e-05  K=1.576e+37


## Mass-class engineering table

The engineering portion the paper quotes. All closed form.

In [4]:
rows = E.run('mass_class_table', {})['result']
cols = ['class','r_s_m','tau_interior_s','T_hawking_K','t_evaporation_yr',
        'kretschmann_core_m^-4','K_core_over_K_Planck','reaches_QGP','echo_delay_s']
print(f"{'class':26s}{'r_s [m]':>12s}{'tau [s]':>12s}{'T_H [K]':>12s}"
      f"{'t_evap [yr]':>13s}{'K_core':>12s}{'K/K_Pl':>12s}{'QGP':>6s}{'echo [s]':>12s}")
for row in rows:
    print(f"{row['class']:26s}{row['r_s_m']:12.2e}{row['tau_interior_s']:12.2e}"
          f"{row['T_hawking_K']:12.2e}{row['t_evaporation_yr']:13.2e}"
          f"{row['kretschmann_core_m^-4']:12.2e}{row['K_core_over_K_Planck']:12.2e}"
          f"{str(row['reaches_QGP']):>6s}{row['echo_delay_s']:12.2e}")

class                          r_s [m]     tau [s]     T_H [K]  t_evap [yr]      K_core      K/K_Pl   QGP    echo [s]
kugelblitz / primordial       1.49e-15    4.95e-24    1.23e+11     2.67e+12    4.93e+60    3.37e-79  True    4.55e-22
stellar (10 M_sun)            2.95e+04    9.85e-05    6.17e-09     2.10e+70    3.15e-17   2.15e-156 False    1.78e-02
intermediate (1e4 M_sun)      2.95e+07    9.85e-02    6.17e-12     2.10e+79    3.15e-29   2.15e-168 False    1.92e+01
supermassive (1e9 M_sun)      2.95e+12    9.85e+03    6.17e-17     2.10e+94    3.15e-49   2.15e-188 False    2.14e+06


## Secondary predictions — allowed to fail, kept in data

`energy_partition` (default split `1 - d* : d*`) and `cosmic_cavitation_budget` (naive `Ω_cav ≈ Ω_BH(1-d*)` vs `Ω_Λ`). The budget is expected to fall short by orders — the signature, if any, is **dark flow** (directional), not a dark-energy magnitude.

In [5]:
print('energy_partition (10 Msun):', E.run('energy_partition', {'M_kg': 10*M.M_SUN})['result'])
print()
print('cosmic_cavitation_budget:', E.run('cosmic_cavitation_budget', {})['result'])

energy_partition (10 Msun): {'space_fraction': 0.754, 'matter_fraction': 0.246, 'E_total_J': 1.7875521500932312e+48, 'E_space_J': 1.3478143211702963e+48, 'E_matter_J': 4.397378289229348e+47, 'basis': '1 - d* (default)'}

cosmic_cavitation_budget: {'omega_bh_assumed': 1e-05, 'space_fraction': 0.754, 'omega_cavitation': 7.540000000000001e-06, 'omega_lambda_obs': 0.6847, 'ratio_to_lambda': 1.1012122097268879e-05, 'verdict': 'falls short — dark-flow (directional) signature, not a dark-energy magnitude'}


## Open items (honest)

- No derivation that nature selects this interior — the engine checks *if*-gravastar-*then*-`K_core` is the de Sitter value; it does not solve a field equation.
- Bounce-time and echo-delay **coefficients** are `O(1)` and model dependent (Haggard–Rovelli; Cardoso–Pani). Only the scalings are firm.
- The `d*` energy split is asserted. `cosmic_cavitation_budget` is expected FALSIFIED as a magnitude.
- Labelled-ZD note: the box-kite object is **PSL(2,7)** (order 168); Moreno's G₂ is the blow-up that forgets the labelling.